packages/react/src/components/Accordion/AccordionTitle.tsx

```diff
@@ -49,37 +70,67 @@ class AccordionTitle extends UIComponent<WithAsProp<AccordionTitleProps>, any> {
   static propTypes = {
     ...commonPropTypes.createCommon(),
     active: PropTypes.bool,
-    index: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
+    buttonRef: PropTypes.object,
+    canBeCollapsed: PropTypes.bool,
+    contentId: PropTypes.string,
+    index: PropTypes.number,
     onClick: PropTypes.func,
     indicator: customPropTypes.itemShorthand,
   }
 
+  static defaultProps = {
+    accessibility: accordionTitleBehavior,
+    as: 'dt',
+  }
+
+  actionHandlers: AccessibilityActionHandlers = {
+    performClick: e => {
+      e.preventDefault()
+      this.handleClick(e)
+    },
+  }
+
   handleClick = e => {
     _.invoke(this.props, 'onClick', e, this.props)
   }
 
-  renderComponent({ ElementType, classes, unhandledProps, styles }) {
-    const { children, content, indicator, active } = this.props
+  handleFocus = (e: React.SyntheticEvent) => {
+    e.stopPropagation()
+    _.invoke(this.props, 'onFocus', e, this.props)
+  }
+
+  renderComponent({ ElementType, classes, unhandledProps, styles, accessibility }) {
+    const { buttonRef, children, content, indicator, active } = this.props
     const indicatorWithDefaults = indicator === undefined ? {} : indicator
 
     const contentElement = (
-      <Layout
-        start={Icon.create(indicatorWithDefaults, {
-          defaultProps: {
-            name: active ? 'stardust-arrow-down' : 'stardust-arrow-end',
-            styles: styles.indicator,
-          },
-        })}
-        main={rtlTextContainer.createFor({ element: content })}
-      />
+      <Ref innerRef={buttonRef}>
+        <Button
+          onFocus={this.handleFocus}
+          {...accessibility.attributes.button}
+          {...applyAccessibilityKeyHandlers(accessibility.keyHandlers.button, unhandledProps)}
+        >
+          <Layout
+            start={Icon.create(indicatorWithDefaults, {
```

Comment:
Why aren't we simple using the icon and content slots on the `Button`?

Reply:
I don't know, that's how it was implemented before. Now I am usign Box instead of Button. Should I refactor it better?

Reply:
I think we would need a plain text variant of a button to make this look neutral. Both button and textOnly button have centered caption for example. I think a div is better